akka-http-core/src/main/scala/akka/http/impl/settings/ConnectionPoolSettingsImpl.scala

```diff
@@ -63,7 +68,9 @@ object ConnectionPoolSettingsImpl extends SettingsCompanion[ConnectionPoolSettin
         case "legacy" ⇒ PoolImplementation.Legacy
         case "new"    ⇒ PoolImplementation.New
       },
-      c getPotentiallyInfiniteDuration "response-entity-subscription-timeout"
+      c getPotentiallyInfiniteDuration "response-entity-subscription-timeout",
+      Map.empty
     )
   }
+
```

Comment:
My thought was that the config override wouldn't be visible in code. Basically I was thinking that if users supply a connection pool setting, that would always be taken as whatever was passed. the config overrides would only be available if you don't pass in an explicit setting (and thus it doesn't make sense to allow configuring the overrides from anywhere other than configuration).

I'm open to changing it, but I personally find it confusing to allow host-specific overrides when you're passing an explicitly configured setting in your code

Reply:
I don't see what you mean with the extremely confusing. Configs should always be possible to be changed using either way since people may have different needs and use cases. Does not matter if we strongly encourage the config way, methods should exist.

Reply:
Hmm, I see what you mean, but actually our convention is to have things modeled in our settings classes. 

For `singleRequest` it also makes sense to create only one object of settings with all the overrides.